### PR TITLE
Fix prefix formatting

### PR DESCRIPTION
**Pull request by theblackcat102**
Move question and answer token to formatting.py and added in dataset phase instead of collator. This fix soda and prosocial dialogue which previously added the prefix as follow:

```
<human><prefix>Your prefix</prefix>prompt question<bot>answer text <eos token>
```

dataset with prefix (soda, prosocial_dialogue) should now be as follow :

```
<prefix>Your prefix</prefix><human>prompt question<bot>answer text <eos token>
```

### CODE REVIEW

File: model/supervised_finetuning/custom_datasets/formatting.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+QA_SPECIAL_TOKENS = {"Question": "<human>", "Answer": "<bot>", "StartPrefix": "<prefix>", "EndPrefix": "</prefix>"}
+
+
+def format_pair(pair):
+    return "{} {} {}".format(QA_SPECIAL_TOKENS["Question"], pair[0], QA_SPECIAL_TOKENS["Answer"]), pair[1]
```
Comment (sanagno) [inline model/supervised_finetuning/custom_datasets/formatting.py:5]: Maybe we can make this into
-+ return "{}{}{}".format(QA_SPECIAL_TOKENS["Question"], pair[0], QA_SPECIAL_TOKENS["Answer"]), pair[1]

Not sure how all the tokenizer handle spaces for the models we have

**Review by sanagno — APPROVED** [review]
ProsocialDialogue and SODA should also have the format_parirs

**Comment by theblackcat102** [comment]
@sanagno I added the prefix token in the preprocess phase, so there's no reason to add during get_item

File: model/supervised_finetuning/custom_datasets/formatting.py (hunk 1 above)
Comment (theblackcat102) [reply model/supervised_finetuning/custom_datasets/formatting.py]: done

**Comment by sanagno** [comment]
Thanks, looks great

**Pull request merged**